libs/mock-builder/src/storage.rs

```diff
@@ -1,64 +1,87 @@
-use std::{any::Any, cell::RefCell, collections::HashMap, fmt};
+use std::{cell::RefCell, collections::HashMap, fmt};
+
+use super::util::TypeSignature;
 
 /// Identify a call in the call storage
 pub type CallId = u64;
 
-trait Callable {
-	fn as_any(&self) -> &dyn Any;
+struct CallInfo {
```

Comment:
Very cool that we can have this as known fixed size, though have comment below in casting thread about the transmute.